### PR TITLE
add natural-string<? and natural-string-ci<?

### PR DESCRIPTION
**Pull request by owaddell**
This pull request adds predicates that compare strings in a way that is more natural to humans.

### CODE REVIEW

File: doc/swish/erlang.tex [hunk 1]
```diff
@@ -2979,6 +2979,29 @@ \subsection{String Utilities}
 \var{separator}. When \var{last-separator} is specified, it is used as
 the last separator.
 
+\pagebreak
```
Comment (burgerrg) [inline doc/swish/erlang.tex]: This manual page break may introduce a maintenance burden.

File: doc/swish/erlang.tex [hunk 2]
```diff
@@ -2979,6 +2979,29 @@ \subsection{String Utilities}
...
+
+These procedures compare strings \var{s$_1$} and \var{s$_2$} in a way that is
+more natural to humans than the corresponding
+\code{string<?} and \code{string-ci<?} predicates.
+The natural comparison predicates:
+\begin{enumerate}
+ \item ignore leading and trailing whitespace
+ \item treat runs of multiple whitespace characters as a single space
+ \item compare integers numerically, treating \code{-} as a dash
+   if it follows an alphanumeric character or else as negation
```
Comment (burgerrg) [inline doc/swish/erlang.tex]: consider using a comma at the end of items 1 & 2, and ", and" at the end of item 3.

**Review by burgerrg — APPROVED** [review]
This looks handy!

File: doc/swish/erlang.tex (hunk 1 above)
Comment (owaddell) [reply doc/swish/erlang.tex]: Good catch. I left that in by mistake.

File: doc/swish/erlang.tex (hunk 2 above)
Comment (owaddell) [reply doc/swish/erlang.tex]: Much better!

**Comment by owaddell** [comment]
Thanks for the speedy review!

**Pull request merged**